### PR TITLE
feat: text search for task/trial logs [DET-7446]

### PR DESCRIPTION
**Pull request by trentwatt**
## Description

add text search to trial logs. [DET-7446]

added support to text search for trial/task logs stored in postgres.

made an attempt to do the same for elastic, but dont know how to test. I tried uncommenting the `elastic` stage in `devcluster.yaml` then starting a task, but the new tasks still ended up using the postgres backend. in any case not all filtering functionalities are implemented for elasticsearch anyways.

https://user-images.githubusercontent.com/28383080/179709684-3d302f9d-b38e-4ac5-9ebf-8e061978990b.mov

## Commentary (optional)

streaming causes the api to get hit pretty frequently. if we think queries might get expensive, we may want to increase the interval.

## Checklist

- [ ] User-facing API changes need the "User-facing API Change" label.
- [ ] Release notes should be added as a separate file under `docs/release-notes/`.
See [Release Note](../docs/release-notes/README.md) for details.
- [ ] Licenses should be included for new code which was copied and/or modified from any external code.

[DET-7446]: https://determinedai.atlassian.net/browse/DET-7446?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by hkang1** [comment]
pretty damn cool!

@stoksc, will this be ok for SLURM and all the other permutations besides elasticsearch?

I almost think this just be the first filtering UI to show (AKA move the search box to the far left) cc @alexjdetermined

File: master/internal/db/postgres_filters.go [hunk 1]
```diff
@@ -60,6 +60,10 @@ func filterToSQL(
 		return fmt.Sprintf("AND %s > $%d", field, paramID)
 	case api.FilterOperationLessThanEqual:
 		return fmt.Sprintf("AND %s <= $%d", field, paramID)
+	case api.FilterOperationStringContainment:
+		// Suprisingly (?) this work for both bytea (needed for current)
+		// as well as plain text fields (not needed atm)
+		return fmt.Sprintf("AND encode(%s::bytea, 'escape') ILIKE  ('%%%%' || $%d || '%%%%')", field, paramID)
```
Comment (trentwatt) [inline master/internal/db/postgres_filters.go:66]: hilarous code

**Review by stoksc — APPROVED** [review]
backend code is a-ok

File: master/internal/db/postgres_tasks.go [hunk 2]
```diff
@@ -266,6 +266,7 @@ type taskLogsFollowState struct {
 func (db *PgDB) TaskLogs(
 	taskID model.TaskID, limit int, fs []api.Filter, order apiv1.OrderBy, followState interface{},
 ) ([]*model.TaskLog, interface{}, error) {
+	fmt.Println("pg logs")
```
Comment (stoksc) [inline master/internal/db/postgres_tasks.go:269]: bump

File: master/internal/db/postgres_filters.go [hunk 3]
```diff
@@ -72,6 +76,9 @@ func filterToParams(f api.Filter) []interface{} {
 		for _, v := range vs {
 			params = append(params, v)
 		}
+	case string:
+		// any reason not to do this?
```
Comment (stoksc) [inline master/internal/db/postgres_filters.go:80]: nope it's fine. prob could rewrite this with generics now.

File: master/internal/db/postgres_filters.go (hunk 1 above)
Comment (hkang1) [reply master/internal/db/postgres_filters.go:65]: [related](https://github.com/determined-ai/determined/pull/4270) to the hilarious code

**Pull request merged**